From d6d47393bab32dc60b43e0eeac16c035000a0358 Mon Sep 17 00:00:00 2001
From: Eugene Popov <popov895@ukr.net>
Date: Sun, 24 Jul 2022 00:18:44 +0000
Subject: [PATCH] [FIX] Unable to remove manually added wallpaper

It's not necessary to assign m_removableWallpapers asynchronously, because the value assigned to it doesn't depend on the result of that asynch function.

!!! This MR is for the Plasma/5.25 branch only and it SHOULD NOT be cherry-picket into the master branch.

BUG: 457019
---
 wallpapers/image/plugin/backgroundlistmodel.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/wallpapers/image/plugin/backgroundlistmodel.cpp b/wallpapers/image/plugin/backgroundlistmodel.cpp
index 8c5630712..0b12470a5 100644
--- a/wallpapers/image/plugin/backgroundlistmodel.cpp
+++ b/wallpapers/image/plugin/backgroundlistmodel.cpp
@@ -107,6 +107,8 @@ void BackgroundListModel::reload(const QStringList &selected)
         return;
     }
 
+    m_removableWallpapers = QSet<QString>(selected.constBegin(), selected.constEnd());
+
     const QStringList dirs = QStandardPaths::locateAll(QStandardPaths::GenericDataLocation, QStringLiteral("wallpapers/"), QStandardPaths::LocateDirectory);
 
     BackgroundFinder *finder = new BackgroundFinder(m_wallpaper.data(), dirs);
@@ -117,7 +119,6 @@ void BackgroundListModel::reload(const QStringList &selected)
         }
 
         processPaths(selected + wallpapersFound);
-        m_removableWallpapers = QSet<QString>(selected.constBegin(), selected.constEnd());
     });
     m_findToken = token;
     finder->start();
-- 
GitLab

